matrix.sh

```diff
@@ -54,7 +54,13 @@ help() {
 }
 
 _curl() {
-	curl -s -H "$AUTHORIZATION" -H "User-Agent: matrix.sh/$VERSION" "$@"
+	HEADERFILE="$HOME/.matrix.sh-headers"
+	if [ -z "$(grep "$AUTHORIZATION" $HEADERFILE)" ] ; then
+		echo "User-Agent: matrix.sh/$VERSION" > $HEADERFILE
```

Comment:
I would do `(umask 0077; echo data > file)` instead of separate chmod, to avoid ever having the secret in a world writable file.